node/api.go

```diff
@@ -342,3 +343,69 @@ func (s *publicWeb3API) ClientVersion() string {
 func (s *publicWeb3API) Sha3(input hexutil.Bytes) hexutil.Bytes {
 	return crypto.Keccak256(input)
 }
+
+type ExecutionPoolSize struct {
+	HttpLimit int
+	WSLimit   int
+}
+
+type ExecutionPoolRequestTimeout struct {
+	HttpLimit time.Duration
+	WSLimit   time.Duration
+}
+
+func (api *privateAdminAPI) GetExecutionPoolSize() *ExecutionPoolSize {
+	httpLimit := api.node.http.httpHandler.Load().(*rpcHandler).server.GetExecutionPoolSize()
+	wsLimit := api.node.ws.wsHandler.Load().(*rpcHandler).server.GetExecutionPoolSize()
+
+	executionPoolSize := &ExecutionPoolSize{
+		HttpLimit: httpLimit,
+		WSLimit:   wsLimit,
+	}
+
+	return executionPoolSize
+}
+
+func (api *privateAdminAPI) GetExecutionPoolRequestTimeout() *ExecutionPoolRequestTimeout {
+	httpLimit := api.node.http.httpHandler.Load().(*rpcHandler).server.GetExecutionPoolRequestTimeout()
+	wsLimit := api.node.ws.wsHandler.Load().(*rpcHandler).server.GetExecutionPoolRequestTimeout()
+
+	executionPoolRequestTimeout := &ExecutionPoolRequestTimeout{
+		HttpLimit: httpLimit,
+		WSLimit:   wsLimit,
+	}
+
+	return executionPoolRequestTimeout
+}
+
+func (api *privateAdminAPI) SetWSExecutionPoolRequestTimeout(n int) *ExecutionPoolRequestTimeout {
+	api.node.ws.wsConfig.executionPoolRequestTimeout = time.Duration(n) * time.Second
+	api.node.ws.wsHandler.Load().(*rpcHandler).server.SetExecutionPoolRequestTimeout(time.Duration(n) * time.Second)
+	log.Warn("Updating ws execution pool request timeout", "timeout", n)
+
+	return api.GetExecutionPoolRequestTimeout()
+}
+
+func (api *privateAdminAPI) SetHttpExecutionPoolRequestTimeout(n int) *ExecutionPoolRequestTimeout {
+	api.node.http.httpConfig.executionPoolRequestTimeout = time.Duration(n) * time.Second
+	api.node.http.httpHandler.Load().(*rpcHandler).server.SetExecutionPoolRequestTimeout(time.Duration(n) * time.Second)
+	log.Warn("Updating http execution pool request timeout", "timeout", n)
+
+	return api.GetExecutionPoolRequestTimeout()
+}
+
+func (api *privateAdminAPI) SetWSExecutionPoolSize(n int) *ExecutionPoolSize {
+	api.node.ws.wsConfig.executionPoolSize = uint64(n)
+	api.node.ws.wsHandler.Load().(*rpcHandler).server.SetExecutionPoolSize(n)
+	log.Warn("Updating ws execution pool size", "threads", n)
+
+	return api.GetExecutionPoolSize()
+}
+
+func (api *privateAdminAPI) SetHttpExecutionPoolSize(n int) *ExecutionPoolSize {
+	api.node.http.httpConfig.executionPoolSize = uint64(n)
```

Comment:
is it atomic? why we need both? direct `=` and `SetExecutionPoolSize` ?

Reply:
One of them is updating the http config which does not need to be atomic. Another one is actually calling the RPC function to set the new size which will actually update the number of workers used by the execution pool. There's a mutex for that in the library.

Reply:
could we have a few SetHttpExecutionPoolSize at the same time?